addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,263 @@
+
+            String token = new StringBuilder("Bearer ")
+                    .append(b.getConfiguration().get(CONFIG_BRIDGE_AUTH_TOKEN).toString()).toString();
+
+            obtainMeterId(token);
+
+            // Start polling job with the interval, that has been set up in the bridge
+            int pollingPeriod = Integer.parseInt(b.getConfiguration().get(CONFIG_BRIDGE_REFRESH).toString());
+            pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    logger.debug("Try to refresh meter data");
+                    try {
+                        updateMeter(token);
+                    } catch (RuntimeException r) {
+                        updateStatus(ThingStatus.OFFLINE);
+                        throw r;
+                    }
+                }
+            }, pollingPeriod, pollingPeriod, TimeUnit.MINUTES);
+            logger.debug("Refresh job scheduled to run every {} hours. for '{}'", pollingPeriod, getThing().getUID());
+
+        } catch (RuntimeException r) {
+            logger.debug("Caught exception in ScheduledExecutorService of BridgeHandler. RuntimeException: ", r);
+        } catch (Exception e) {
+            logger.debug("Could not initialize Thing {}: ", getThing().getUID(), e);
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("Bridge Status updated to {} for device: {}", bridgeStatusInfo.getStatus(), getThing().getUID());
+        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, bridgeStatusInfo.getDescription());
+        }
+
+    }
+
+    /**
+     * Requests the corresponding meter data and stores the meterId internally for later usage
+     *
+     * @param token The current active auth token
+     */
+    private void obtainMeterId(String token) {
+        try {
+
+            String url = getApiString(API_METER_ENDPOINT);
+
+            Properties urlHeader = new Properties();
+            urlHeader.put("CONTENT-TYPE", "application/json");
+            urlHeader.put("Authorization", token);
+
+            String urlResponse = HttpUtil.executeUrl("GET", url, urlHeader, null, null, 2000);
+            JsonObject responseJson = (JsonObject) new JsonParser().parse(urlResponse);
+
+            if (responseJson.has("meter_id")) {
+                setMeterID(responseJson.get("meter_id").toString());
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+
+            String errorMsg = new StringBuilder("Invalid Api Response ( ").append(responseJson.toString()).append(" )")
+                    .toString();
+
+            throw new Exception(errorMsg);
```

Comment:
Same comment about usage not throw Exception but if no specific Exception known make it `RuntimeException`